src/layers/IdentityCoordinatesTileLayer.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-param-reassign */
+
+/*
+------------------------------------------
+THIS CODE IS FROM DECKGL MOSTLY
+IT IS AWAITING A PR THERE AND WILL
+THEN BE REMOVED IN FAVOR OF THAT
+------------------------------------------
+*/
+
+import { TileLayer } from 'deck.gl';
+import TileCache from './tile-layer-utils/TileCache';
+
+export class IdentityCoordinatesTileLayer extends TileLayer {
+  initializeState() {
+    this.state = {
+      tiles: [],
+      isLoaded: false,
+    };
+  }
+
+  shouldUpdateState({ changeFlags }) {
+    return changeFlags.somethingChanged;
+  }
+
+  // passsing in maxHeight and maxWidth to the TileCache
+  updateState({ props, context, changeFlags }) {
+    let { tileCache } = this.state;
+    if (
+      !tileCache
+      || (changeFlags.updateTriggersChanged
+        && (changeFlags.updateTriggersChanged.all
+          || changeFlags.updateTriggersChanged.getTileData))
+    ) {
+      const {
+        getTileData, maxZoom, minZoom, maxCacheSize,
+      } = props;
+      if (tileCache) {
+        tileCache.finalize();
+      }
+      tileCache = new TileCache({
+        getTileData,
+        maxSize: maxCacheSize,
+        maxZoom,
+        minZoom,
+        maxHeight: props.maxHeight,
+        maxWidth: props.maxWidth,
+        tileSize: props.tileSize,
+        onTileLoad: this._onTileLoad.bind(this),
+        onTileError: this._onTileError.bind(this),
+      });
+      this.setState({ tileCache });
+    } else if (changeFlags.updateTriggersChanged) {
+      // if any updateTriggersChanged (other than getTileData), delete the layer
+      this.state.tileCache.tiles.forEach((tile) => {
+        tile.layer = null;
+      });
+    }
+    const { viewport } = context;
+    if (changeFlags.viewportChanged && viewport.id !== 'DEFAULT-INITIAL-VIEWPORT') {
+      const z = this.getLayerZoomLevel();
+      tileCache.update(viewport);
+      // The tiles that should be displayed at this zoom level
+      const currTiles = tileCache.tiles.filter(tile => tile.z === z);
+      this.setState({ isLoaded: false, tiles: currTiles });
+      this._onTileLoad();
```

Comment:
I believe `setState` is asynchronous in React, so `_onTileLoad` will likely not be called using the updated `tiles` state. You can pass `_onTileLoad` as a callback in `setState`.

On a broader note, what is the purpose of using both `tiles` and `tileCache` in state? I see the cache is filtered both here and in the rendering function.

Reply:
Sorry, I meant `setState` in deck-gl's `TileLayer`. Here's an example: https://github.com/uber/deck.gl/commit/4db9528022b48f4cfa46330f5002a0be6edb0ab7

Reply:
This well may be true, but this was a direct pull from deck.gl's code (see note at the top).  I will bring this up with them when I make the PR for this stuff against their codebase.

Reply:
Ok, I see. Thanks for clarifying.